app/src/test/java/com/android/unio/model/firestore/HydrationAndSerializationTest.kt

```diff
@@ -114,18 +116,8 @@ class HydrationAndSerializationTest {
     assertEquals(association.name, serialized["name"])
     assertEquals(association.fullName, serialized["fullName"])
     assertEquals(association.description, serialized["description"])
-    assertEquals(
-        association.members.associate { member -> member.user.uid to member.role.uid },
-        serialized["members"])
-    assertEquals(
-        association.roles.associate { role ->
-          role.uid to
-              mapOf(
-                  Role::displayName.name to role.displayName,
-                  Role::permissions.name to
-                      role.permissions.getGrantedPermissions().map { it.stringName })
-        },
-        serialized["roles"])
+    assertEquals(mapUsersToRoles(association.members), serialized["members"])
+    assertEquals(mapRolesToPermission(association.roles), serialized["roles"])
```

Comment:
It's great to put the mapping in a separate function, makes the test class clearer :)